packages/shared/src/components/feeds/FeedSettings/sections/FeedSettingsContentPreferencesSection.tsx

```diff
@@ -65,9 +65,11 @@ export const FeedSettingsContentPreferencesSection = (): ReactElement => {
                 videoSetting.defaultEnabledState
               }
               onToggleCallback={() =>
-                onToggleSettings(
-                  videoSetting.id,
-                  videoSetting.defaultEnabledState,
+                editFeedSettings(() =>
+                  onToggleSettings(
+                    videoSetting.id,
+                    videoSetting.defaultEnabledState,
+                  ),
```

Comment:
Since some of the settings we make to feed are async and saved immediately we were not following or toggling `isDirty` flag for form when user would select those. 

Idea with `editFeedSettings` wrapper is to wrap or call this method whenever we do some async change to feed. For DX purposes it is also possible to wrap the original method in it and it will just be called. 

The method then toggles dirty flag and we can use it to prompt user for saving changes when closing/canceling feed creation.